src/replication/log_consumer.hpp

```diff
@@ -28,22 +28,27 @@
 
 #include "cubstream.hpp"
 #include "semaphore.hpp"
-#include "slave_control_channel.hpp"
+#include "stream_entry_fetcher.hpp"
 #include "thread_manager.hpp"
-#include <chrono>
-#include <condition_variable>
 #include <cstddef>
 #include <memory>
-#include <queue>
 
 namespace cubthread
 {
   class daemon;
 };
 
+namespace cubreplication
+{
+  class applier_worker_task;
+  class stream_entry;
+  class subtran_applier;
+}
+
 namespace cubstream
 {
   class multi_thread_stream;
+  using repl_stream_entry_fetcher = entry_fetcher<cubreplication::stream_entry>;
```

Comment:
What do you think is better cubstream::repl_stream_entry_fetcher or cubreplication::stream_entry_fetcher?